app/src/main/java/com/example/ithaca_transit_android_v2/MainActivity.kt

```diff
@@ -1,50 +1,57 @@
 package com.example.ithaca_transit_android_v2
 
+import android.content.Context
+import android.content.pm.PackageManager
+import android.location.LocationManager
 import android.os.Bundle
-import android.util.Log
-import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 import com.example.ithaca_transit_android_v2.models.Location
+import com.example.ithaca_transit_android_v2.presenters.MapPresenter
 import com.example.ithaca_transit_android_v2.presenters.SearchPresenter
 import com.example.ithaca_transit_android_v2.ui_adapters.SearchViewAdapter
+
+import com.example.ithaca_transit_android_v2.util.CurrLocationManager
+
 import com.google.android.gms.maps.GoogleMap
 import com.google.android.gms.maps.OnMapReadyCallback
 import com.google.android.gms.maps.SupportMapFragment
 import io.reactivex.disposables.Disposable
 import kotlinx.android.synthetic.main.activity_main.*
-import kotlinx.android.synthetic.main.activity_toolbar_search.*
+
+import kotlinx.android.synthetic.main.search_main.*
+import kotlinx.android.synthetic.main.search_secondary.*
+
 
 class MainActivity : AppCompatActivity(), OnMapReadyCallback {
     private lateinit var searchDisposable: Disposable
     private var mSearchLocations: List<Location> = ArrayList()
     private lateinit var mSearchAdapter: SearchViewAdapter
     private lateinit var mSearchPresenter: SearchPresenter
+    private lateinit var mCurrLocationManager: CurrLocationManager
 
     override fun onMapReady(map: GoogleMap?) {
-        map!!.setOnMapClickListener { point ->
-            Log.i("qwerty", "map clicked")
-            search_view.clearFocus()
-        }
+        val p = MapPresenter()
+        p.initMapView(map!!)
```

Comment:
should we add a catch for null maps, ie when there is no wifi? we can display a pop-up that says "no internet" or display a no wifi sign

Reply:
Added TODO